### PR TITLE
Allow users to specify RuntimeEnv

### PR DESCRIPTION
**Pull request by omus**
Extracted the `RuntimeEnv` portion of #38. Requires #31 to be squashed first.

### CODE REVIEW

**Review by kleinschmidt — APPROVED** [review]
this looks goo

File: Ray.jl/src/runtime.jl [hunk 1]
```diff
@@ -112,7 +112,16 @@ function submit_task(f::Function, args...)
...
+
+    # Generate the JSON representation of `RuntimeEnvInfo`:
+    # https://github.com/ray-project/ray/blob/ray-2.5.1/src/ray/protobuf/runtime_env_common.proto#L40-L41
+    serialize_runtime_env_info = JSON3.write(
+        Dict(
+            "serialized_runtime_env" => JSON3.write(json_dict(runtime_env))::String
+        )
+    )
+
+    return GC.@preserve args rayjll._submit_task(fd, object_ids, serialize_runtime_env_info)
```
Comment (omus) [inline Ray.jl/src/runtime.jl:124]:
```suggestion
    serialized_runtime_env_info = JSON3.write(
        Dict(
            "serialized_runtime_env" => JSON3.write(json_dict(runtime_env))::String
        )
    )

    return GC.@preserve args rayjll._submit_task(fd, object_ids, serialized_runtime_env_info)
```

File: src/wrappers/any.jl [hunk 2]
```diff
@@ -150,10 +150,10 @@ end
 
 # XXX: Need to convert julia vectors to StdVector. This function helps us avoid having
 # CxxWrap as a direct dependency in Ray.jl
-function _submit_task(dir, fd, oids::AbstractVector)
+function _submit_task(fd, oids::AbstractVector, serialize_runtime_env_info)
```
Comment (omus) [inline src/wrappers/any.jl:153]:
```suggestion
function _submit_task(fd, oids::AbstractVector, serialized_runtime_env_info)
```

File: src/wrappers/any.jl [hunk 3]
```diff
@@ -150,10 +150,10 @@ end
...
     # https://github.com/JuliaInterop/CxxWrap.jl/issues/367
     args = isempty(oids) ? StdVector{ObjectID}() : StdVector(oids)
-    return _submit_task(dir, fd, args)
+    return _submit_task(fd, args, serialize_runtime_env_info)
```
Comment (omus) [inline src/wrappers/any.jl:156]:
```suggestion
    return _submit_task(fd, args, serialized_runtime_env_info)
```

**Pull request merged**